sources/core/Stride.Core/Serialization/Binary/BinarySerialization.cs

```diff
@@ -11,6 +12,7 @@ namespace Stride.Core.Serialization
     /// <remarks>
     /// This class is a simple front end to <see cref="BinarySerializationReader"/> and <see cref="BinarySerializationWriter"/>.
     /// </remarks>
+    [Obsolete] // not used, inefficient
```

Comment:
I'm pretty sure this is the class users would use in their code to use the serialization system - it should not be deprecated unless a better backwards compatible alternative is provided.

Reply:
It is silly to new up two objects in some two line static method which returns neither of them. Shouldn't they be requesting some kind of service from DI instead? Or they should be directed to just create the reader or writer themselves imho.

Reply:
Ok, fair enough. Since the change isn't very complicated for the user conceptually let's go with your recommendation in the Obsolete attribute. Usually you just need a simple static [de]serialization method and this what provided here.